service/storage_service.cc

```diff
@@ -5747,12 +5747,20 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(shar
                         // Bootstrap did not complete yet, but streaming did
                     } else {
                         co_await retrier(_bootstrap_result, coroutine::lambda([&] () ->future<> {
+                            assert(_topology_state_machine._topology.req_param.contains(raft_server.id()));
                             if (is_repair_based_node_ops_enabled(streaming::stream_reason::replace)) {
-                                co_await _repair.local().replace_with_repair(get_token_metadata_ptr(), rs.ring.value().tokens, {});
+                                std::unordered_set<gms::inet_address> ignored_ips;
+                                for (const auto& id : std::get<replace_param>(_topology_state_machine._topology.req_param[raft_server.id()]).ignored_ids) {
+                                    auto ip = _group0->address_map().find(id);
```

Comment:
Please live a FIXME with a reference to #6403 (we should not need to do this translation here)